src/appleseed-max-impl/appleseedenvmap/appleseedenvmap.cpp

```diff
@@ -0,0 +1,588 @@
+        m_pblock->GetValue(ParamIdSatMultiplier, t, m_sat_multiplier, m_params_validity);
+        m_pblock->GetValue(ParamIdHorizonShift, t, m_horizon_shift, m_params_validity);
+        m_pblock->GetValue(ParamIdGroundAlbedo, t, m_ground_albedo, m_params_validity);
+
+        NotifyDependents(FOREVER, PART_ALL, REFMSG_CHANGE);
+    }
+
+    if (!m_map_validity.InInterval(t))
+	{
+		m_map_validity.SetInfinite();
+	    if (m_turbidity_map)
+	        m_turbidity_map->Update(t,m_map_validity);
+	}
+
+    valid &= m_map_validity;
+    valid &= m_params_validity;
+}
+
+void AppleseedEnvMap::Reset()
+{
+    m_params_validity.SetEmpty();
+    m_map_validity.SetEmpty();
+}
+
+Interval AppleseedEnvMap::Validity(TimeValue t)
+{
+    Interval valid = FOREVER;
+    Update(t, valid);
+    return valid;
+}
+
+ParamDlg* AppleseedEnvMap::CreateParamDlg(HWND hwMtlEdit, IMtlParams* imp)
+{
+    IAutoMParamDlg* masterDlg = g_appleseed_appleseedenvmap_classdesc.CreateParamDlgs(hwMtlEdit, imp, this);
+    g_block_desc.SetUserDlgProc(new EnvMapParamMapDlgProc());
+
+    return masterDlg;
+}
+
+const USHORT ChunkFileFormatVersion                 = 0x0001;
+const USHORT ChunkMtlBase                           = 0x1000;
+const USHORT FileFormatVersion = 0x0001;
+
+IOResult AppleseedEnvMap::Save(ISave* isave)
+{
+    bool success = true;
+
+    isave->BeginChunk(ChunkFileFormatVersion);
+    success &= write(isave, &FileFormatVersion, sizeof(FileFormatVersion));
+    isave->EndChunk();
+
+    isave->BeginChunk(ChunkMtlBase);
+    success &= MtlBase::Save(isave) == IO_OK;
+    isave->EndChunk();
+
+    return success ? IO_OK : IO_ERROR;
+}
+
+IOResult AppleseedEnvMap::Load(ILoad* iload)
+{
+    IOResult result = IO_OK;
+
+    while (true)
+    {
+        result = iload->OpenChunk();
+        if (result == IO_END)
+            return IO_OK;
+        if (result != IO_OK)
+            break;
+
+        switch (iload->CurChunkID())
+        {
+          case ChunkFileFormatVersion:
+            {
+                USHORT version;
+                result = read<USHORT>(iload, &version);
+            }
+            break;
+
+          case ChunkMtlBase:
+            result = MtlBase::Load(iload);
+            break;
+        }
+
+        if (result != IO_OK)
+            break;
+
+        result = iload->CloseChunk();
+        if (result != IO_OK)
+            break;
+    }
+
+    return result;
+}
+
+AColor AppleseedEnvMap::EvalColor(ShadeContext& sc)
+{
+    Color basecolor (0.13f, 0.58f, 1.0f);
+    if (!sc.InMtlEditor())
+        return basecolor;
+
+    //render gradient for the thumbnail
+    Point3 p = sc.UVW(0);
+    Color white (1.0f, 1.0f, 1.0f);
+
+	return AColor((basecolor * (float)p.y) + (white * (float)(1.0-p.y)));
+}
+
+Point3 AppleseedEnvMap::EvalNormalPerturb(ShadeContext& /*sc*/)
+{
+    return Point3(0, 0, 0);
+}
+
+foundation::auto_release_ptr<renderer::EnvironmentEDF> AppleseedEnvMap::create_envmap(const char* name)
```

Comment:
Traditionally we defined `asf` and `asr` namespace aliases and use that, it makes the code shorter...